mkosi/__init__.py

```diff
@@ -311,25 +405,77 @@ def gpt_root_native(arch: Optional[str], usr_only: bool = False) -> GPTRootTypeT
         arch = platform.machine()
 
     if usr_only:
-        if arch in ("i386", "i486", "i586", "i686"):
+        if arch == "alpha":
+            return GPTRootTypeTriplet(GPT_USR_ALPHA, GPT_USR_ALPHA_VERITY, GPT_USR_ALPHA_VERITY_SIG)
+        elif arch == "sparc":
+            return GPTRootTypeTriplet(GPT_USR_ARC, GPT_USR_ARC_VERITY, GPT_USR_ARC_VERITY_SIG)
+        elif arch == "armv7l":
+            return GPTRootTypeTriplet(GPT_USR_ARM, GPT_USR_ARM_VERITY, GPT_USR_ARM_VERITY_SIG)
+        elif arch == "aarch64":
+            return GPTRootTypeTriplet(GPT_USR_ARM64, GPT_USR_ARM64_VERITY, GPT_USR_ARM64_VERITY_SIG)
+        elif arch == "ia64":
+            return GPTRootTypeTriplet(GPT_USR_IA64, GPT_USR_IA64_VERITY, GPT_USR_IA64_VERITY_SIG)
+        elif arch == "mips":
+            return GPTRootTypeTriplet(GPT_USR_MIPS_LE, GPT_USR_MIPS_LE_VERITY, GPT_USR_MIPS_LE_VERITY_SIG)
```

Comment:
Hmm, I think this isn't good. The arch is usually called mipsel, to distinguish it from the big-endian variant. Please use the same identifiers that systemd uses.

(Unfortunately the identifiers in systemd are not the standard ones. For example, "mips-le" instead of the usual "mipsel". I think we should at some point change them, but we need to do it in a backwards-compatible fashion, so it's not so easy.)

Reply:
Didn't know we had `uname_architecture()` right in the systemd tree. Updated to use the names from there

Reply:
> (Unfortunately the identifiers in systemd are not the standard ones. For example, "mips-le" instead of the usual "mipsel". I think we should at some point change them, but we need to do it in a backwards-compatible fashion, so it's not so easy.)

if there was a "standard" that'd be great. Unfortunately noone can agree on anything. i.e. amd64 is what debian calls x86-64 and so on...

go calls "mipsle" what debian calls "mipsel" which we call "mips-le"... (see https://github.com/golang/go/commit/ad366fdbe40eea7f4fbdc75e8a27304c696793f1)

Reply:
Yeah, but Debian has been calling it mipsel already before go was a thing… So I don't see a reason to not follow the same naming in 2016.

Reply:
I think this was done to be systematic, i.e. whenever there are two endianess flavours of some arch we used the _LE and _BE suffixes, to distinguish them.